File: typescript/design-as-you-code/domain/CarWaterSystem.ts
import Logger from "../shared/Logger";
import Car from "./Car";

const SWIPE_TIMES_AFTER_WASHING = 3;

export default class CarWaterSystem {
    constructor(private car: Car) {}

    async washFrontal() {
        if(!this.car.turnedOn){
            return;
        }

        Logger.log('Washing frontal...');

        await this.car.windshieldWiper.activateTemporarily(SWIPE_TIMES_AFTER_WASHING);

        Logger.log('Washed frontal');
    }
}

File: python/src/domain/carWaterSystem.py
from src.shared.logger import log

SWIPE_TIMES_AFTER_WASHING = 3

class CarWaterSystem:
    def __init__(self, car):
        self._car = car

    async def washFrontal(self):
        if not self._car.turnedOn:
            return
        
        log('Washing frontal...')

        await self._car.windshieldWiper.activateTemporarily(SWIPE_TIMES_AFTER_WASHING)

        log('Washed frontal')